src/documents/IoCAPI.md

```diff
@@ -0,0 +1,133 @@
+component and returns its path as a set of parsed array segments. This information can be very useful for making calculations about
+the _geometry_ of component trees - that is, which components are descended from which others and which are siblings, etc.
+
+### fluid.queryIoCSelector(root, selector[, flat])
+
+Query for all components matching a selector in a particular tree.
+* `root {Component}` The root component at which to start the search. A reasonable choice for this can be [`fluid.rootComponent`](#fluid-rootcomponent) although
+such global searches can be very slow. A more reasonable choice will be some particular component in the tree whose descendents are of interest.  
+* `selector {String}` An [IoCSS selector](IoCSS.md), in form of a string. Note that since selectors supplied to this function implicitly
+match downwards, they need not contain the "head context" followed by whitespace required in the distributeOptions form. E.g.
+simply `fluid.viewComponent` will match all viewComponents below the root.
+* `flat {Boolean}` [optional] `true` if the search should just be performed at top level of the component tree
+ Note that with `flat=true` this search will scan every component in the tree and may well be very slow.
+
+### fluid.expand(material, that)
+
+[Expands](ExpansionOfComponentOptions.md) some options material in the context of a particular component - any [IoC References](IocReferences.md) and expanders held at any
+depth in the material will be expanded to hold their resolved contents. The source `material` will not be modified.
+
+* `material {Object}` The configuration material to be expanded
+* `that {Component}` The component in whose [context](Contexts.md) the material is to be expanded
+* Returns: `{Object}` An expanded version of the input `material`.
+
+### fluid.getForComponent(component, path)
+
+This is a fairly interesting method. During the instantiation of an IoC component tree, the framework's "focus of attention" moves around the tree
+in a data-driven way. If an [IoC reference](#IoCReferences.md) is seen to a particular path, this will direct attention to instantiate the
+material referenced by the path. `fluid.getForComponent` is the method used internally by the framework in order to ensure that any referenced
+material has actually been instantiated before the results from any IoC reference are filled in in the [expanded](ExpansionOfComponentOptions.md)
+options material. You can use this method too, in order to make sure that any member of a component has been instantiated before you look at it. This
+should not normally be necessary - and the use of this API only makes sense during a construction "fit" - during the "semi-static period", you can
+assume that all members of all visible components are concrete. 
+
+This is an analogue of the API [`fluid.get`](CoreAPI.md#fluid-get-model-path-) for use during the "ginger construction process.
```

Comment:
The reader knows nothing about the ginger construction process, so it's probably not helpful to refer to it.